### DISCUSSION TITLE
Branch protection rules: Require a pull request before merging does not stop commits that come from editing the README

### DISCUSSION THREAD

**Discussion by zephyap** [community/community#19793] [Pull Requests] [answered]
Had an interesting occurrence today where a commit was made to `master` in one of my repositories despite the "Require a pull request before merging" being active. I looked at it and it was a change to the README file (still problematic because now my QA branch and master branch have differing histories). It turns out that the commit was made via the "Edit this file" button that appears when viewing the repository through the browser.

Has anyone encountered this before? It feels unintuitive that commits can essentially still be made directly to a protected branch, and imo is something we would want to change.

**Comment by EtheWallet** [comment]
f2cbf3a6b8b731ca76f68154783f2c7c977a11da2165ff2c21eca5e43f00f56c

**Comment by zephyap** [comment, marked as answer]
Update: Turned out to be a user who had admin permissions, which I was not previously aware of
